tianshou/data/utils/converter.py

```diff
@@ -107,21 +113,32 @@ def to_hdf5(x: Hdf5ConvertibleType, y: h5py.Group) -> None:
             subgrp.attrs["__data_type__"] = v.__class__.__name__
             to_hdf5(subgrp_data, subgrp)
         elif isinstance(v, (np.ndarray, torch.Tensor)):
```

Comment:
If you keep this condition as is, I don't see why `y.create_dataset(k, data=to_numpy(v))` could fail.

Reply:
One problem are arrays with `dtype=object`. For example, if `None` appears in the info from the gym environment, this happens as Trinkle23897 pointed out. Possibly there are also other cases which are not directly supported by HDF5 like structured arrays. To treat all of them explicitly might be difficult.

Reply:
Ok I see. Why not then. Just add a comment for us to remember the initial motivation of catching it.

Reply:
After rethinking about it, I don't think such behavior can happen with `torch.Tensor`. They are way more restrictive than Numpy array, so just catching the exception for numpy.ndarray should be fine and help to make the exception more clear.

Reply:
Ok, agreed. I have split the elif statement up into two cases for Tensors and ndarrays.